### PR TITLE
Update to JDK8 as a minimum

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

Everyone should have moved on to Java8+, let's make this the minimum so we can remove some workarounds and make use of default methods etc.

Modifications:

Update to Java8

Result:

Be able to use Java8 features and remove workarounds

### CODE REVIEW

**Comment by normanmaurer** [comment]
@He-Pin @chrisvest @violetagg @ejona86 @trustin let's get this in first and then we do a followup pr to change things to remove workaround etc.

This will be the base for 4.2.0.Final

**Comment by He-Pin** [comment]
Epic !

File: pom.xml [hunk 1]
```diff
@@ -649,8 +613,8 @@
   </profiles>
 
   <properties>
-    <maven.compiler.source>1.6</maven.compiler.source>
-    <maven.compiler.target>1.6</maven.compiler.target>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (He-Pin) [inline pom.xml:617]: What about compile with Java 11 but with a release target 8？

Maybe that will cause some problems with no such method exception at runtime.

Comment (He-Pin) [reply pom.xml:617]: That will need update ci too

**Comment by He-Pin** [comment]
return majorVersion(SystemPropertyUtil.get("java.specification.version", "1.6"));

Some code like this will need an update too.

File: pom.xml (hunk 1 above)
Comment (violetagg) [reply pom.xml:617]: the docker should be using already 1.8?

**Review by He-Pin — APPROVED** [review]
Lgtm, but will need to update the codebase for minimal java version checking too, that can be done later, this is Epic, thanks.

**Review by violetagg — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (normanmaurer) [reply pom.xml:617]: Yes it uses 1.8 already

**Pull request merged**